packages/material-ui/src/Rating/Rating.js

```diff
@@ -366,6 +357,8 @@ const Rating = React.forwardRef(function Rating(props, ref) {
       ref={handleRef}
       onMouseMove={handleMouseMove}
       onMouseLeave={handleMouseLeave}
+      onClick={readOnly ? null : handleClick}
+      onKeyPress={readOnly ? null : handleClick}
```

Comment:
This was really just to avoid an eslint warning, but could also potentially be a noop (`() => null`). The `handleClear` handler was already ignoring keyboard events that were triggering `onClick`